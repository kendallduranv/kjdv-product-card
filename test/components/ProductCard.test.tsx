import React from 'react';

import renderer from 'react-test-renderer'
import { ProductCard } from '../../src/components/index';
import { product1 } from '../data/products';


const { act } = renderer;


describe('ProductCard', () => {
    test('debe de mostrar el componente correctamente en el titulo personalizado', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });


    test('debe de incrementat el constador', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(1)}></button>
                        </>
                    )
                }
            </ProductCard>
        );

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();

        })


        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1')

    })

})

