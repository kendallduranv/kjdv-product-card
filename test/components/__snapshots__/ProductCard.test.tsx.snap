// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard debe de incrementat el constador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard debe de mostrar el componente correctamente en el titulo personalizado 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug -Card  sin imagen
  </span>
</div>
`;

exports[`ProductTitle debe de mostrar el componente correctamente en el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
